triqs/arrays/impl/indexmap_storage_pair.hpp

```diff
@@ -170,6 +170,7 @@ namespace triqs { namespace arrays {
 
     auto const & memory_layout() const { return indexmap_.memory_layout(); }
     indexmap_type const & indexmap() const {return indexmap_;}
+    indexmap_type & indexmap() {return indexmap_;}
```

Comment:
I don't think its a good idea to expose the indexmap via a non-const reference.
The few functions that need actually change indexmap_ should be friend functions.

Reply:
It is a bad idea, I agree, but how would you implement `set_mem_layout_or_check_if_view()` then?
Making it a friend of `array`, `matrix` and `vector` isn't a great solution either.